### PR TITLE
Update XFEM mesh on initialization

### LINKED ISSUE #6503: Perform XFEM mesh modification on initialization
### Description of the enhancement or error report
Currently, XFEM only modifies the mesh at the end of the timestep, and then repeats the solve if needed.  Add a call to modify the mesh at initialization.

### Rationale for the enhancement or information for reproducing the error
For cases with a stationary crack, this is an important performance improvement, especially if only one step is solved, because it avoids solving the problem without the cut before solving with it.  This will also be important down the road when we use XFEM for material interfaces.

### Identified impact
Small modification to FEProblem.C, mods to XFEM classes

### PR DESCRIPTION
**Pull request by bwspenc**
closes #6503

### CODE REVIEW

File: modules/xfem/src/base/XFEMGeometricCut2D.C [hunk 1]
```diff
@@ -102,54 +111,57 @@ bool
...
   Real tol = 1.e-15;
 
   const Real fraction = cutFraction(time);
 
-  unsigned int n_sides = frag_edges.size();
-
-  for (unsigned int i = 0; i < n_sides; ++i)
+  if (fraction > 0.0)
   {
-    Real seg_originx = frag_edges[i][0](0);
```
Comment (bwspenc) [inline modules/xfem/src/base/XFEMGeometricCut2D.C]: Note that the duplication of code here that appeared above will be addressed by #6505.

File: modules/xfem/src/base/XFEM.C [hunk 2]
```diff
@@ -339,53 +339,59 @@ XFEM::markCutEdgesByGeometry(Real time)
 {
   bool marked_edges = false;
 
-  MeshBase::element_iterator       elem_it  = _mesh->elements_begin();
```
Comment (bwspenc) [inline modules/xfem/src/base/XFEM.C]: This is for you, @permcody!

File: modules/xfem/src/base/XFEM.C [hunk 3]
```diff
@@ -339,53 +339,59 @@ XFEM::markCutEdgesByGeometry(Real time)
...
 
-    // mark cut edges for the element and its fragment
-    for (unsigned int i=0; i<_geometric_cuts.size(); ++i)
-    {
-      _geometric_cuts[i]->cutElementByGeometry(elem, elem_cut_edges, time);
-      if (CEMElem->numFragments() > 0)
-        _geometric_cuts[i]->cutFragmentByGeometry(frag_edges, frag_cut_edges, time);
-    }
+      // mark cut edges for the element and its fragment
+      for (unsigned int i=0; i<active_geometric_cuts.size(); ++i)
```
Comment (permcody) [inline modules/xfem/src/base/XFEM.C]: spacing

File: modules/xfem/src/base/XFEMGeometricCut2D.C [hunk 4]
```diff
@@ -24,65 +24,74 @@ XFEMGeometricCut2D::~XFEMGeometricCut2D()
...
+  return cutFraction(time) > 0;
+}
+
+bool
 XFEMGeometricCut2D::cutElementByGeometry(const Elem* elem, std::vector<CutEdge> & cut_edges, Real time)
 {
-  //Use the algorithm described here to determine whether edges are cut by the cut line:
-  //http://stackoverflow.com/questions/563198/how-do-you-detect-where-two-line-segments-intersect
+  // Use the algorithm described here to determine whether edges are cut by the cut line:
+  // http://stackoverflow.com/questions/563198/how-do-you-detect-where-two-line-segments-intersect
```
Comment (permcody) [inline modules/xfem/src/base/XFEMGeometricCut2D.C]: Seem like we have some potential code duplication here:
https://github.com/idaholab/moose/blob/devel/framework/src/utils/LineSegment.C

Comment (bwspenc) [inline modules/xfem/src/base/XFEMGeometricCut2D.C]: Yes, see my note above.  I have an issue for this (#6505) that I plan to resolve shortly after this gets merged.

**Comment by permcody** [comment]
:+1:

**Pull request merged**